src/main/flight/mixer_init.c

```diff
@@ -396,6 +396,11 @@ static void mixerConfigureOutput(void)
 }
 #endif // USE_QUAD_MIXER_ONLY
 
+float getDynamicIdleOffset(void)
```

Comment:
This is the offset of the DShot (digital) idle value, not the dynamic idle offset.
It is present whether or not dynamic idle is active.
Maybe use `getDigitalIdleOffset`?

Reply:
There's a subtle difference - if dynamic idle is used the offset is applied to the throttle range, so it is scaled with `PWM_RANGE`. If dynamic idle is not used, the offset is applied to the motor output range.
I'll see if I can come up with something that covers both.

Reply:
#10317 makes it a bit better - but it's still ugly as the logic is now in the driver level. What is really needed is that higher level logic like offset / range calculation is pulled up from the driver level, maybe into `io/dshot.c` / `io/motor.c`.

Reply:
I've rebased my [mixer changes PR 10294](https://github.com/betaflight/betaflight/pull/10294) to this PR, and realised a good way of resolving that offset issue, so that when dynamic idle is active, the offset is only a value below which throttle can go, not an offset applied across the full range.  This works fabulously well and solves the ugly hack I previously used.  
I painstakingly checked all the interactions of each throttle and motor modifier and they are all now very, very good.  And it all works great with the log viewer changes, too.
There remain some very subtle imperfections, perhaps relating to scaling over PWM range, but overall I think it's looking much better than ever now.